File: domains/algorithms/implementation/angry-professor/angry-professor.cpp
/*
 * angry-professor.cpp
 * https://www.hackerrank.com/challenges/angry-professor
 *
 * Solution by mabernardo
 * https://github.com/mabernardo
 * https://www.hackerrank.com/mabernardo
 *
 */
#include <iostream>
#include <vector>
using namespace std;

bool cancelled(vector<int>& arr, int threshold)
{
	int ontime = 0;
	for (const auto &i : arr)
	{
		if (i <= 0)
			++ontime;
	}
	return ontime < threshold;
}

int main()
{
	int tests;
	cin >> tests;
	while (tests--)
	{
		int students;
		int threshold;
		vector<int> arrivals;

		cin >> students;
		cin >> threshold;

		while (students--)
		{
			int a;
			cin >> a;
			arrivals.push_back(a);
		}

		if (cancelled(arrivals, threshold))
			cout << "YES\n";
		else
			cout << "NO\n";
	}
}

File: domains/algorithms/implementation/angry-professor/AngryProfessor.java
/*
 * AngryProfessor.java
 * https://www.hackerrank.com/challenges/angry-professor
 *
 * Solution by mabernardo
 * https://github.com/mabernardo
 * https://www.hackerrank.com/mabernardo
 *
 */
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class AngryProfessor {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int tests = scan.nextInt();
		
		while (tests-- > 0) {
			int students = scan.nextInt();
			int threshold = scan.nextInt();
			
			List<Integer> arrivals = new ArrayList<>();
			while (students-- > 0) {
				Integer a = scan.nextInt();
				arrivals.add(a);
			}
			
			if (hasCancelled(arrivals, threshold)) {
				System.out.println("YES");
			} else {
				System.out.println("NO");
			}
		}
		scan.close();
	}
	
	private static boolean hasCancelled(List<Integer> list, int threshold) {
		int ontime = 0;
		for (Integer i : list) {
			if (i <= 0) {
				++ontime;
			}
		}
		return ontime < threshold;
	}
}
